src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotTest.java

```diff
@@ -945,6 +946,75 @@ public void verifyConnectorFailsIfMarkSnapshotDoneFails(boolean consistentSnapsh
         YugabyteDBSnapshotChangeEventSource.FAIL_WHEN_MARKING_SNAPSHOT_DONE = false;
     }
 
+    @Test
+    public void snapshotShouldBeCompletedOnParentIfSplitHappenedAfterStreamCreation() throws Exception {
+        /*
+        * The objective of the test is to verify that snapshot takes place on the parent tablet if a split
+        * happens after a consistent stream is created and before connector has been deployed. Additionally,
+        * any DMLs performed after the split are not part of snapshot records.
+         */
+        TestHelper.dropAllSchemas();
+        TestHelper.execute("CREATE TABLE t1 (id INT PRIMARY KEY, name TEXT) SPLIT INTO 1 TABLETS;");
+        TestHelper.execute("CREATE TABLE t2 (id INT PRIMARY KEY, name TEXT) SPLIT INTO 1 TABLETS;");
+        TestHelper.execute("CREATE TABLE t3 (id INT PRIMARY KEY, name TEXT) SPLIT INTO 1 TABLETS;");
+
+        int recordsCount = 10;
+        String insertFormat = "INSERT INTO t1 VALUES (%d, 'value for split table');";
+        for (int i = 0; i < recordsCount; ++i) {
+            TestHelper.execute(String.format(insertFormat, i));
+        }
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "t1", false, true, true, true);
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
+        configBuilder.with(YugabyteDBConnectorConfig.SNAPSHOT_MODE, "initial");
+
+        YBClient ybClient = TestHelper.getYbClient(getMasterAddress());
+        YBTable table = TestHelper.getYbTable(ybClient, "t1");
+
+        // Verify that there is just a single tablet.
+        Set<String> tablets = ybClient.getTabletUUIDs(table);
+        int tabletCountBeforeSplit = tablets.size();
+        assertEquals(1, tabletCountBeforeSplit);
+
+        // Compact the table to ready it for splitting.
+        ybClient.flushTable(table.getTableId());
+
+        // Wait for 20s for the table to be flushed.
+        TestHelper.waitFor(Duration.ofSeconds(20));
+
+        // Split the tablet. There is just one tablet so it is safe to assume that the iterator will
+        // return just the desired tablet.
+        ybClient.splitTablet(tablets.iterator().next());
+
+        // Wait till there are 2 tablets for the table.
+        TestHelper.waitForTablets(ybClient, table, 2);
+
+        // Insert 5 more rows.
```

Comment:
Add a comment related to the test:

```
// Insert 5 more rows, since these records will be inserted after the split,
// they should not be consumed at the time of consumption of snapshot from parent tablet.
```